src/screens/TopicPageScreen/index.js

```diff
@@ -413,31 +425,35 @@ const TopicPageScreen = (props) => {
         isFollow={isFollow}
         getTopicDetail={getTopicDetail}
       />
-      <TiktokScroll
-        ref={listRef}
-        contentHeight={dimen.size.TOPIC_CURRENT_ITEM_HEIGHT + normalizeFontSizeByWidth(4)}
-        data={feeds}
-        onEndReach={onEndReach}
-        onRefresh={onRefresh}
-        refreshing={loading}
-        renderItem={renderItem}
-        onScroll={handleScroll}
-        scrollEventThrottle={16}
-        showSearchBar={true}
-        searchHeight={isHeaderHide ? headerHideHeight : headerShowHeight}
-        snap
-        contentOffset={{x: 0, y: topPosition}}
-        contentInsetAdjustmentBehavior={feeds?.length > 1 ? 'automatic' : 'never'}
-        onMomentumScrollEnd={(event) => onWillSendViewPostTime(event, feeds)}
-      />
+
+      <View
+        style={{
+          marginTop: isHeaderHide ? -dimen.size.TOPIC_FEED_HEADER_HEIGHT : 0
+        }}>
+        <TiktokScroll
+          ref={listRef}
+          contentHeight={dimen.size.TOPIC_CURRENT_ITEM_HEIGHT + normalizeFontSizeByWidth(4)}
+          data={feeds}
+          onEndReach={onEndReach}
+          onRefresh={onRefresh}
+          refreshing={loading}
+          renderItem={renderItem}
+          onScroll={handleScroll}
+          scrollEventThrottle={16}
+          showSearchBar={true}
+          snap
+          contentInsetAdjustmentBehavior={feeds?.length > 1 ? 'automatic' : 'never'}
+          onMomentumScrollEnd={(event) => onWillSendViewPostTime(event, feeds)}
+        />
+      </View>
       <ButtonAddPostTopic topicName={topicName} onRefresh={onRefresh} />
       <BlockComponent
         ref={refBlockComponent}
         refresh={onBlockCompleted}
         refreshAnonymous={onDeleteBlockedPostCompleted}
         screen="topic_screen"
       />
-    </SafeAreaProvider>
+    </View>
   );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes in this section mainly involve wrapping the `TiktokScroll` component inside a `View` and adjusting its `marginTop` based on the `isHeaderHide` state. This could potentially improve the layout and appearance of the page. However, it's important to ensure that the `isHeaderHide` state is correctly managed to prevent layout issues. Also, the removal of `searchHeight` and `contentOffset` props from `TiktokScroll` might affect the functionality of the search bar and the initial scroll position of the list. If these features are still required, you might need to reintroduce these props.

```diff
+ <View
+   style={{
+     marginTop: isHeaderHide ? -dimen.size.TOPIC_FEED_HEADER_HEIGHT : 0
+   }}>
+   <TiktokScroll
+     ref={listRef}
+     contentHeight={dimen.size.TOPIC_CURRENT_ITEM_HEIGHT + normalizeFontSizeByWidth(4)}
+     data={feeds}
+     onEndReach={onEndReach}
+     onRefresh={onRefresh}
+     refreshing={loading}
+     renderItem={renderItem}
+     onScroll={handleScroll}
+     scrollEventThrottle={16}
+     showSearchBar={true}
+     snap
+     contentInsetAdjustmentBehavior={feeds?.length > 1 ? 'automatic' : 'never'}
+     onMomentumScrollEnd={(event) => onWillSendViewPostTime(event, feeds)}
+   />
+ </View>
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->